diff --git a/ohos/notice/collect_module_notice_file.py b/ohos/notice/collect_module_notice_file.py
--- a/ohos/notice/collect_module_notice_file.py
+++ b/ohos/notice/collect_module_notice_file.py
@@ -59,12 +59,12 @@ def find_other_files(license_file_path):
 
 
 def find_file_recursively(current_dir: str, target_files: list):
-    if is_top_dir(current_dir):
-        return None
     for file in target_files:
         candidate = os.path.join(current_dir, file)
         if os.path.isfile(candidate):
             return candidate
+    if is_top_dir(current_dir):
+        return None
     return find_file_recursively(os.path.dirname(current_dir), target_files)
 
 
diff --git a/ohos/notice/collect_system_notice_files.py b/ohos/notice/collect_system_notice_files.py
--- a/ohos/notice/collect_system_notice_files.py
+++ b/ohos/notice/collect_system_notice_files.py
@@ -77,7 +77,6 @@ def collect_notice_files(options, dest_dir: str, depfiles: str):
                 generate_notice_files(dest_dir, module_info, depfiles)
 
     notice_files = build_utils.get_all_files(options.notice_root_dir)
-    depfiles.extend(notice_files)
     for file in notice_files:
         dest = os.path.join(dest_dir,
                             os.path.relpath(file, options.notice_root_dir))
