diff --git a/hb/util/loader/load_ohos_build.py b/hb/util/loader/load_ohos_build.py
--- a/hb/util/loader/load_ohos_build.py
+++ b/hb/util/loader/load_ohos_build.py
@@ -787,12 +787,16 @@ def process_other_parts(part_name, parts_config_dict, parts_info_output_path):
         parts_info_file = os.path.join(parts_info_output_path,
                                        part_name + ".json")
         if part_name == 'hisysevent_config':
+            check_changes = True
             parts_info_file = os.path.join(parts_info_output_path,
                                            'hisysevent_configs' + ".json")
+        else:
+            check_changes = False
         if part_name == 'parts_inner_kits_info':
             parts_info_file = os.path.join(parts_info_output_path,
                                            'inner_kits_info' + ".json")
-        write_json_file(parts_info_file, parts_info)
+        write_json_file(parts_info_file, parts_info,
+                        check_changes=check_changes)
         LogUtil.hb_info("generate '{}' info to '{}'".format(part_name,
                                                             parts_info_file), mode=Config.log_mode)
 
diff --git a/ohos/hisysevent/hisysevent_process.py b/ohos/hisysevent/hisysevent_process.py
--- a/ohos/hisysevent/hisysevent_process.py
+++ b/ohos/hisysevent/hisysevent_process.py
@@ -101,12 +101,14 @@ def main(argv):
                                                     src_part_list,
                                                     args.source_root_dir,
                                                     args.gn_root_build_dir)
+    depfiles.extend(hisysevent_config_files)
 
     _output_path = os.path.dirname(args.hisysevent_install_info_file)
     install_info_list = _get_install_info(hisysevent_config_files,
                                           _output_path,
                                           args.config_install_dest_dir)
-    write_json_file(args.hisysevent_install_info_file, install_info_list)
+    write_json_file(args.hisysevent_install_info_file, install_info_list,
+                    check_changes=True)
     build_utils.write_depfile(args.depfile, args.hisysevent_install_info_file,
                               depfiles)
 
