diff --git a/ohos/generate_part_info.py b/ohos/generate_part_info.py
--- a/ohos/generate_part_info.py
+++ b/ohos/generate_part_info.py
@@ -104,13 +104,16 @@ def gen_output_file(part_name: str, origin_part_name: str, all_modules_file: str
     part_install_modules.extend(thirdparty_module_list)
     part_install_modules.extend(hdi_module_list)
     # write install modules file
-    write_json_file(install_modules_file, part_install_modules)
+    write_json_file(install_modules_file, part_install_modules,
+                    check_changes=True)
 
     # add no install modules to dict, example: host target
     modules_info_dict[part_name] = part_no_install_modules
     # write dep modules file
-    write_json_file(dep_modules_file, modules_info_dict)
-    write_json_file(output_host_file, part_host_modules)
+    write_json_file(dep_modules_file, modules_info_dict,
+                    check_changes=True)
+    write_json_file(output_host_file, part_host_modules,
+                    check_changes=True)
 
 
 def main():
diff --git a/ohos/packages/BUILD.gn b/ohos/packages/BUILD.gn
--- a/ohos/packages/BUILD.gn
+++ b/ohos/packages/BUILD.gn
@@ -11,6 +11,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import("//build/config/ohos/copy_ex.gni")
 import("//build/config/python.gni")
 import("//build/ohos/sa_profile/sa_profile.gni")
 
@@ -161,18 +162,20 @@ foreach(_platform, target_platform_list) {
 
   _notice_root_dir = "${current_platform_dir}/NOTICE_FILES"
   _collected_notice_zipfile = "${current_platform_dir}/system_notice_files.zip"
+  _staged_notice_zipfile =
+      "$target_gen_dir/system_notice_files_${_platform}.zip"
 
-  action_with_pydeps("collect_notice_files__${_platform}") {
+  action_with_pydeps("stage_notice_files__${_platform}") {
     deps = [ ":${_platform}_parts_list" ]
     script = "//build/ohos/notice/collect_system_notice_files.py"
     depfile = "$target_gen_dir/$target_name.d"
     sources = [ _system_install_info_file ]
-    outputs = [ _collected_notice_zipfile ]
+    outputs = [ _staged_notice_zipfile ]
     args = [
       "--install-info-file",
       rebase_path(_system_install_info_file, root_build_dir),
       "--output-file",
-      rebase_path(_collected_notice_zipfile, root_build_dir),
+      rebase_path(_staged_notice_zipfile, root_build_dir),
       "--depfile",
       rebase_path(depfile, root_build_dir),
       "--notice-root-dir",
@@ -180,6 +183,14 @@ foreach(_platform, target_platform_list) {
     ]
   }
 
+  copy_ex("collect_notice_files__${_platform}") {
+    deps = [ ":stage_notice_files__${_platform}" ]
+    dest = current_platform_dir
+    renaming_sources = [ _staged_notice_zipfile ]
+    renaming_destinations = [ "system_notice_files.zip" ]
+    outputs = [ _collected_notice_zipfile ]
+  }
+
   _merged_notice_txt = "${current_platform_dir}/NOTICE.txt"
   _notice_module_info_file = "${current_platform_dir}/NOTICE_module_info.json"
   action_with_pydeps("merge_system_notice_file_${current_platform}") {
diff --git a/ohos/packages/parts_install_info.py b/ohos/packages/parts_install_info.py
--- a/ohos/packages/parts_install_info.py
+++ b/ohos/packages/parts_install_info.py
@@ -69,7 +69,8 @@ def main():
     platform_out_dir = os.path.dirname(args.system_install_info_file)
     if not os.path.exists(platform_out_dir):
         os.makedirs(platform_out_dir, exist_ok=True)
-    write_json_file(args.system_install_info_file, _parts_list)
+    write_json_file(args.system_install_info_file, _parts_list,
+                    check_changes=True)
 
     if args.depfile:
         _dep_files = []
diff --git a/scripts/util/file_utils.py b/scripts/util/file_utils.py
--- a/scripts/util/file_utils.py
+++ b/scripts/util/file_utils.py
@@ -16,7 +16,6 @@
 import json
 import os
 import subprocess
-import hashlib
 import platform
 
 
@@ -82,18 +81,9 @@ def write_json_file(output_file, content, check_changes=False):
 
 
 def __check_changes(output_file, content):
-    if os.path.exists(output_file) and os.path.isfile(output_file):
-        # file content md5 val
-        sha256_obj = hashlib.sha256()
-        sha256_obj.update(str(read_json_file(output_file)).encode())
-        hash_value = sha256_obj.hexdigest()
-        # new content md5 val
-        sha256_obj_new = hashlib.sha256()
-        sha256_obj_new.update(str(content).encode())
-        hash_value_new = sha256_obj_new.hexdigest()
-        if hash_value_new == hash_value:
-            return False
-    return True
+    if not os.path.isfile(output_file):
+        return True
+    return read_json_file(output_file) != content
 
 
 # Write file data
