diff --git a/ohos/notice/collect_system_notice_files.py b/ohos/notice/collect_system_notice_files.py
index f86585fe7ebeb7f3b1c93fe95a0b61e51efd6977..b5c4db7ad4a968d9cc9313d9a1e8aa41dcf22cb6 100755
--- a/ohos/notice/collect_system_notice_files.py
+++ b/ohos/notice/collect_system_notice_files.py
@@ -77,7 +77,6 @@ def collect_notice_files(options, dest_dir: str, depfiles: str):
                 generate_notice_files(dest_dir, module_info, depfiles)
 
     notice_files = build_utils.get_all_files(options.notice_root_dir)
-    depfiles.extend(notice_files)
     for file in notice_files:
         dest = os.path.join(dest_dir,
                             os.path.relpath(file, options.notice_root_dir))
